src/radical/pilot/agent/lm/spark.py

```diff
@@ -109,7 +108,7 @@ def lrms_config_hook(cls, name, cfg, lrms, logger):
         	master_ip = lrms.node_list[0]
         else:
         	try:
-        		master_ip = subprocess.check_call('hostname -f')
+        		master_ip = subprocess.check_output('hostname -f'.split()).strip()
```

Comment:
There is no need to split 'hostname -f'. ```check_output``` accepts a list of inputs. I propose it to be ```['hostname','-f']```